File: backend/app/schemas/notification.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.models.enums import NotificationChannel, NotificationType


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    type: NotificationType
    channel: NotificationChannel
    title: str
    message: str
    is_read: bool
    related_entity_type: Optional[str] = None
    related_entity_id: Optional[uuid.UUID] = None
    created_at: datetime


class NotificationUnreadCount(BaseModel):
    count: int


class NotificationMarkAllRead(BaseModel):
    updated: int

File: frontend/src/types/notification.ts
export type NotificationType = "task" | "reminder" | "subscription" | "event" | "system" | "document" | "vehicle"
export type NotificationChannel = "in_app" | "email" | "push"

export interface Notification {
  id: string
  type: NotificationType
  channel: NotificationChannel
  title: string
  message: string
  is_read: boolean
  related_entity_type: string | null
  related_entity_id: string | null
  created_at: string
}
